src/test/conftest.py

```diff
@@ -54,3 +74,416 @@ def create_tables():
+def _unpacking_lock_manager() -> UnpackingLockManager:
+    yield UnpackingLockManager()
+
+
+@pytest.fixture
+def analysis_scheduler(
+    request,
+    pre_analysis_queue,
+    post_analysis_queue,
+    analysis_finished_event,
+    analysis_finished_counter,
+    _unpacking_lock_manager,
+) -> AnalysisScheduler:
+    """Returns an instance of :py:class:`~scheduler.analysis.AnalysisScheduler`.
+    The scheduler has some extra testing features. See :py:class:`SchedulerTestConfig` for the features.
+    """
+    test_config = merge_markers(request, 'SchedulerTestConfig', SchedulerTestConfig.get_subclass_from_request(request))
+
+    # FIXME Don't patch
+    # The reason for patching is documented in :py:func:`AnalysisScheduler.start`.
+    with MonkeyPatch.context() as mkp:
+        mkp.setattr(AnalysisScheduler, '_load_plugins', lambda _: None)
+        _analysis_scheduler = AnalysisScheduler(
+            pre_analysis=lambda _: None,
+            post_analysis=lambda *_: None,
+            unpacking_locks=_unpacking_lock_manager,
+        )
+
+    _analysis_scheduler.db_backend_service = test_config.backend_db_class()
+
+    def _pre_analysis_hook(fw):
+        pre_analysis_queue.put(fw)
+        if test_config.pipeline:
+            _analysis_scheduler.db_backend_service.add_object(fw)
+
+    _analysis_scheduler.pre_analysis = _pre_analysis_hook
+
+    def _post_analysis_hook(*args):
+        post_analysis_queue.put(args)
+        analysis_finished_counter.value += 1
+        # We use == here insead of >= to not set the thing when items_to_analyze is 0
+        if analysis_finished_counter.value == test_config.items_to_analyze:
+            analysis_finished_event.set()
+
+        if test_config.pipeline:
+            _analysis_scheduler.db_backend_service.add_analysis(*args)
+
+    # test_unpack_and_analyse.py
+    _analysis_scheduler.post_analysis = _post_analysis_hook
+
+    def _instanciate_plugin_wo_starting(plugin_class):
+        with MonkeyPatch.context() as mkp:
+            mkp.setattr(plugin_class, 'start', lambda _: None)
+
+        return plugin_class
+
+    # FIXME This is a hack
+    # We want to load plugins but don't actually start them
+    # Due to a bug which is documented in AnalysisScheduler._load_plugins we have to do it really wired
+    with MonkeyPatch.context() as mkp:
+        if not test_config.start_processes:
+            mkp.setattr(_analysis_scheduler, '_instanciate_plugin', _instanciate_plugin_wo_starting)
+        _analysis_scheduler._load_plugins()
+
+    if test_config.start_processes:
+        _analysis_scheduler.start()
+
+    yield _analysis_scheduler
+    # TODO scope: Maybe get inspired by the database_interface fixture.
+    # Have a module scoped thing, then have a function scoped thing that makes sure that all queues are reset.
+    # This would prevent calling exec and fork for every test function
+    # But we have to make sure that the config is the one set for the very specific function in
```

Comment:
Removed the comment as the current code behaves the same as status quo since `setUp()` is called before any test function.But we should probably implement what this comment does: Only start the scheduler once.